kernel/cPrimVal.mli

```diff
@@ -0,0 +1,90 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+type ('constr, 'types, 'univs) t =
+  | Int of Uint63.t
+  | Float of Float64.t
+  | Array of 'univs * 'constr array * 'constr * 'types
+
+val fold :
+  ('a -> 'constr -> 'a) ->
+  ('a -> 'types -> 'a) ->
+  'a ->
+  ('constr, 'types, 'univs) t ->
+  'a
```

Comment:
Would benefit from some documentation comment (same for al other functions in the file).